sources/packages/backend/workflow/test/2021-2022/parttime-assessment/awards-amounts/parttime-assessment-2021-2022-awards-amount-CSGD.e2e-spec.ts

```diff
@@ -0,0 +1,216 @@
+      offeringWeeksAmount,
+      calculatedAssessment.variables.dmnPartTimeAwardAllowableLimits
+        .limitAwardCSGD2OrLessChildAmount,
+    );
+    expect(
+      Math.round(
+        calculatedAssessment.variables.federalAwardCSGDAmount * 10000,
+      ) / 10000,
+    ).toBe(
+      Math.min(
+        maxComparisonCalculation,
+        calculatedAssessment.variables.dmnPartTimeAwardAllowableLimits
+          .limitAwardCSGDAmount,
+      ),
+    );
+  });
+
+  it("Should determine federalAwardCSGDAmount for less than 3 dependants and calculatedDataTotalFamilyIncome <= limitAwardCSGDIncomeCap", async () => {
+    // Arrange
+    const assessmentConsolidatedData =
+      createFakeConsolidatedPartTimeData(PROGRAM_YEAR);
+    assessmentConsolidatedData.studentDataCRAReportedIncome = 20000;
+    assessmentConsolidatedData.studentDataRelationshipStatus = "married";
+    assessmentConsolidatedData.studentDataIsYourSpouseACanadianCitizen =
+      YesNoOptions.Yes;
+    assessmentConsolidatedData.partner1CRAReportedIncome = 20000;
+    // Act
+    const calculatedAssessment = await executePartTimeAssessmentForProgramYear(
+      PROGRAM_YEAR,
+      assessmentConsolidatedData,
+    );
+    // Assert
+    // calculatedDataTotalFamilyIncome <= limitAwardCSGDIncomeCap
+    // federalAwardCSGDAmount
+    expect(
+      calculatedAssessment.variables.calculatedDataTotalFamilyIncome,
+    ).toBeLessThan(
+      calculatedAssessment.variables.dmnPartTimeAwardFamilySizeVariables
+        .limitAwardCSGDIncomeCap,
+    );
+    expect(calculatedAssessment.variables.federalAwardCSGDAmount).toBe(
+      Math.min(
+        calculatedAssessment.variables.dmnPartTimeAwardAllowableLimits
+          .limitAwardCSGD2OrLessChildAmount,
+        calculatedAssessment.variables.dmnPartTimeAwardAllowableLimits
+          .limitAwardCSGDAmount,
+      ),
+    );
+  });
+
+  it("Should determine federalAwardCSGDAmount for less than 3 dependants and calculatedDataTotalFamilyIncome > limitAwardCSGDIncomeCap", async () => {
+    // Arrange
+    const assessmentConsolidatedData =
+      createFakeConsolidatedPartTimeData(PROGRAM_YEAR);
+    assessmentConsolidatedData.studentDataCRAReportedIncome = 30000;
+    assessmentConsolidatedData.studentDataRelationshipStatus = "married";
+    assessmentConsolidatedData.studentDataIsYourSpouseACanadianCitizen =
+      YesNoOptions.Yes;
+    assessmentConsolidatedData.partner1CRAReportedIncome = 30000;
+    // Act
+    const calculatedAssessment = await executePartTimeAssessmentForProgramYear(
+      PROGRAM_YEAR,
+      assessmentConsolidatedData,
+    );
+    // Assert
+    // calculatedDataTotalFamilyIncome > limitAwardCSGDIncomeCap
+    // federalAwardCSGDAmount
+    expect(
+      calculatedAssessment.variables.calculatedDataTotalFamilyIncome,
+    ).toBeGreaterThan(
+      calculatedAssessment.variables.dmnPartTimeAwardFamilySizeVariables
+        .limitAwardCSGDIncomeCap,
+    );
+    const nestedCalculation = Math.max(
```

Comment:
A comment outside the PR and for the future, I have been seeing this logic in the asserts trying to replicate in some way the logic in the workflow. I would recommend that when we have more control over all the data we should try to assert the values in the E2E in a more hard-coded way.
@dheepak-aot @andrepestana-aot @guru-aot @sh16011993 thoughts?

Reply:
I agree. we should not re-write the logic for assertion.

Reply:
I agree too. We should assert hard-coded values instead of the logic.